app/helpers/common_parameters_helper.rb

```diff
@@ -70,9 +70,38 @@ def lookup_key_field(id, value, options)
 
   def authorized_resource_parameters(resource, type)
     parameters_by_type = resource.send(type)
-    parameter_ids_to_view = parameters_by_type.authorized(:view_params).map(&:id)
-    resource_parameters = parameters_by_type.select { |p| parameter_ids_to_view.include?(p.id) }
+    @allowed_ids_hash = find_allowed_param_ids_per_action(parameters_by_type)
+    resource_parameters = parameters_by_type.select { |p| @allowed_ids_hash[:ids_to_view].include?(p.id) }
     resource_parameters += parameters_by_type.select(&:new_record?)
     resource_parameters
   end
+
+  def can_edit_parameter?(parameter)
+    action_allowed?(parameter, @allowed_ids_hash[:ids_to_edit])
+  end
+
+  def can_destroy_parameter?(parameter)
+    action_allowed?(parameter, @allowed_ids_hash[:ids_to_destroy])
+  end
+
+  private
+
+  def find_allowed_param_ids_per_action(parameters_by_type)
+    result = { ids_to_view: parameters_by_type.authorized(:view_params).map(&:id) }
+    user = User.current
+    return result if user.nil?
+    return result if user.admin?
+    resource_class = parameters_by_type.klass
+    result.merge(
+      ids_to_edit: authorizer.cached_collection(resource_class, :edit_params).map(&:id),
+      ids_to_destroy: authorizer.cached_collection(resource_class, :destroy_params).map(&:id)
+    )
+  end
+
+  def action_allowed?(parameter, allowed_ids)
+    user = User.current
```

Comment:
If we use `#authorized` scope, it wil automatically create `1=0` or `1=1` queries for these special cases. We shouldn't worry about it here.